crates/rpc-types/src/eth/transaction/request.rs

```diff
@@ -238,6 +242,137 @@ impl From<Transaction> for TransactionRequest {
+            max_fee_per_gas: Some(tx.max_fee_per_gas),
+            max_priority_fee_per_gas: Some(tx.max_priority_fee_per_gas),
+            gas: Some(tx.gas_limit),
+            value: Some(tx.value),
+            input: TransactionInput::from(tx.input),
+            nonce: Some(tx.nonce),
+            chain_id: Some(tx.chain_id),
+            transaction_type: Some(2),
+            access_list: Some(tx.access_list),
+            ..Default::default()
+        }
+    }
+}
+
+impl From<TxEip4844> for TransactionRequest {
+    fn from(tx: TxEip4844) -> TransactionRequest {
+        TransactionRequest {
+            from: None,
+            to: Some(tx.to),
+            max_fee_per_blob_gas: Some(tx.max_fee_per_blob_gas),
+            gas: Some(tx.gas_limit),
+            max_fee_per_gas: Some(tx.max_fee_per_gas),
+            max_priority_fee_per_gas: Some(tx.max_priority_fee_per_gas),
+            value: Some(tx.value),
+            input: TransactionInput::from(tx.input),
+            nonce: Some(tx.nonce),
+            chain_id: Some(tx.chain_id),
+            transaction_type: Some(3),
+            access_list: Some(tx.access_list),
+            blob_versioned_hashes: Some(tx.blob_versioned_hashes),
+            ..Default::default()
+        }
+    }
+}
+
+impl From<TxEip4844WithSidecar> for TransactionRequest {
+    fn from(tx: TxEip4844WithSidecar) -> TransactionRequest {
+        let sidecar = tx.sidecar;
+        let tx = tx.tx;
+        TransactionRequest {
+            from: None,
+            to: Some(tx.to),
+            max_fee_per_blob_gas: Some(tx.max_fee_per_blob_gas),
+            gas: Some(tx.gas_limit),
+            max_fee_per_gas: Some(tx.max_fee_per_gas),
+            max_priority_fee_per_gas: Some(tx.max_priority_fee_per_gas),
+            value: Some(tx.value),
+            input: TransactionInput::from(tx.input),
+            nonce: Some(tx.nonce),
+            chain_id: Some(tx.chain_id),
+            transaction_type: Some(3),
+            access_list: Some(tx.access_list),
+            blob_versioned_hashes: Some(tx.blob_versioned_hashes),
+            sidecar: Some(sidecar),
+            ..Default::default()
+        }
+    }
+}
+
+impl From<TxEip4844Variant> for TransactionRequest {
+    fn from(tx: TxEip4844Variant) -> TransactionRequest {
+        match tx {
+            TxEip4844Variant::TxEip4844(tx) => tx.into(),
+            TxEip4844Variant::TxEip4844WithSidecar(tx) => tx.into(),
+        }
+    }
+}
+
+impl From<TypedTransaction> for TransactionRequest {
+    fn from(tx: TypedTransaction) -> TransactionRequest {
+        match tx {
+            TypedTransaction::Legacy(tx) => tx.into(),
+            TypedTransaction::Eip2930(tx) => tx.into(),
+            TypedTransaction::Eip1559(tx) => tx.into(),
+            TypedTransaction::Eip4844(tx) => tx.into(),
+        }
+    }
+}
+
+impl From<TxEnvelope> for TransactionRequest {
+    fn from(envelope: TxEnvelope) -> TransactionRequest {
+        match envelope {
```

Comment:
we could attempt to extract `from` from the signature here (ignoring any error on the ecdsa recovery)

Reply:
it does make the UX cumbersome, but I also don't like dropping information that we definitely have and the user is likely to want if that makes sense

I wonder if it would make sense to extend the signed struct with `signed_by: OnceLock<Address>` which can be computed if K256 enabled

Reply:
Regardless, this decision shouldn't block the PR